molgenis-data-rest/src/main/java/org/molgenis/data/rest/v2/RestControllerV2.java

```diff
@@ -299,7 +323,52 @@ public EntityCollectionBatchCreateResponseBodyV2 createEntities(@PathVariable("e
 			response.setStatus(HttpServletResponse.SC_NO_CONTENT);
 			throw e;
 		}
+	}
+
+	/**
+	 * Copy an entity.
+	 * 
+	 * @param entityName
+	 *            name of the entity that will be copied.
+	 * @param request
+	 *            CopyEntityRequestV2
+	 * @param response
+	 *            HttpServletResponse
+	 * @return String name of the new entity
+	 * @throws Exception
+	 */
+	@RequestMapping(value = "copy/{entityName}", method = POST, produces = APPLICATION_JSON_VALUE)
+	@ResponseBody
+	public String copyEntity(@PathVariable("entityName") String entityName,
+			@RequestBody @Valid CopyEntityRequestV2 request, HttpServletResponse response) throws Exception
+	{
+		// No repo
+		if (!dataService.hasRepository(entityName)) throw createUnknownEntityException(entityName);
+
+		// Does the entity already exists
+		if (dataService.hasRepository(request.getNewEntityName())) throw createDuplicateEntityException(request
+				.getNewEntityName());
+
+		Repository repositoryToCopy = dataService.getRepository(entityName);
 
+		// Permission
+		boolean readPermission = permissionService.hasPermissionOnEntity(repositoryToCopy.getName(), Permission.READ);
+		if (!readPermission) throw createNoReadPermissionOnEntityException(entityName);
+
+		// Capabilities
+		boolean writableCapabilities = dataService.getCapabilities(repositoryToCopy.getName()).contains(
+				RepositoryCapability.WRITABLE);
+		if (!writableCapabilities) throw createNoWriteCapabilitiesOnEntityException(entityName);
+
+		// Copy
+		Repository repository = dataService.copyRepositoryRunAsSystem(repositoryToCopy, request.getNewEntityName(),
```

Comment:
If you put the copyRunAsSystem in a separate method here that runs as system, wouldn't that be a bit cleaner?